Remotion/Validation/Core/Validators/MaximumLengthValidator.cs

```diff
@@ -15,18 +15,68 @@
 // along with re-motion; if not, see http://www.gnu.org/licenses.
 // 
 using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
 using JetBrains.Annotations;
+using Remotion.FunctionalProgramming;
+using Remotion.Utilities;
 using Remotion.Validation.Implementation;
+using Remotion.Validation.Results;
 
 namespace Remotion.Validation.Validators
 {
-  public class MaximumLengthValidator : LengthValidator
+  public class MaximumLengthValidator : IMaxLengthValidator
   {
-    public MaximumLengthValidator (int max, [NotNull] ValidationMessage validationMessage)
-        : base(0, max, $"The value must have at most {max} characters.", validationMessage)
+    public int? Max { get; }
+
+    public string ErrorMessage { get; }
+
+    public ValidationMessage ValidationMessage { get; }
+
+    public MaximumLengthValidator (int? max, [NotNull] ValidationMessage validationMessage)
```

Comment:
Max was origianlly not-nullable. please update and simplify.

Reply:
As I checked, I think it was originally nullable: d7e3df2474b420b3832928556936d5237e77a3d1